raft/raft.go

```diff
@@ -390,11 +400,15 @@ func (r *raft) bcastAppend() {
 
 // bcastHeartbeat sends RPC, without entries to all the peers.
 func (r *raft) bcastHeartbeat() {
+	r.bcastHeartbeatWithCtx(nil)
```

Comment:
originally i want to make the first impl simple. So we only need to have an ack count. now we need to know where the ack coming from since dup ack is possible now.